zipline/finance/commission.py

```diff
@@ -108,3 +111,135 @@ def calculate(self, transaction):
         """
         cost_per_share = transaction.price * self.cost
         return cost_per_share, abs(transaction.amount) * cost_per_share
+
+
+class PerShareWithOptions(object):
+    """
+    Calculates a commission for a transaction based on a per
+    share cost with an optional minimum cost per trade.
+    Applies different set of fees to equities and underliers.
+    """
+
+    def __init__(self, equity_cost=DEFAULT_PER_SHARE_COST,
+                 equity_min_trade_cost=DEFAULT_MINIMUM_COST_PER_TRADE,
+                 option_costs=None,
+                 option_min_trade_cost=DEFAULT_MINIMUM_COST_PER_TRADE,
+                 equity_sids=None, option_sids=None):
+        """
+        Args:
+            equity_cost: cost of an equity trade per-share
+            equity_min_trade_cost: the minimum trade cost for equities
+            option_costs: dictionary specifying per-share commissions in the
+                          option commission tiers defined by IB.
+                          For a description of IB option commission tiers, see
+                          the link below. The dictionary is keyed by a 2d
+                          tuple containing the minimum and maximum option
+                          premium. Values of the dict are the per-contract
+                          commission for an option with premium between these
+                          two minimum and maximum values. For example, the
+                          following dict:
+                                tier_dict = {}
+                                tier_dict[(0.10,  np.inf)] = 0.70
+                                tier_dict[(0.05,    0.10)] = 0.50
+                                tier_dict[(-np.inf, 0.05)] = 0.25
+                          indicates that an option with a premium >= $0.10
+                          has a per-contract commission of $0.70. An option
+                          with a premium between $0.05 and $0.10 has a
+                          commission of $0.50 per-contract, and an option
+                          with a premium less than $0.05 has a per-contract
+                          commission of $0.25.
+            option_min_trade_cost: minimum trade cost for options
+            equity_sids: list of integer sids to apply equity commissions
+            option_sids: list of integer sids to apply option commissions
```

Comment:
Ideally we could just use the type of the `Asset` object to know which type of commission to apply. Right now we only have `Future` and `Equity`. I think it might not be too hard to just add this class for use here.

Another option would to accept only `tiered_assets` as a set of a `Asset` objects to apply the tiered pricing to, maybe with some sentinel that says, "all of them". We might want to say `tiered_assets` instead of ``option_sids` because we should represent if something is an option or not at the type level.

I will explain the choice of set over list below.